1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,65 @@
-function setAlarm() {}
+let countDown;
+let [min, sec] = [0, 0];
+const pauseBtnParentNode = document.querySelector("#set").parentNode;
+const pauseBtn = document.createElement("button");
+pauseBtn.id = "pause";
+pauseBtn.type = "button";
+pauseBtn.textContent = "Pause";
+pauseBtnParentNode.append(pauseBtn);
+pauseBtn.addEventListener("click", () => {
+  clearInterval(countDown);
+});
+function flashedbg() {
+  const bodyStyle = document.body.style;
+  const black = "rgb(37, 37, 37)";
+  bodyStyle.backgroundColor = black;
+  bodyStyle.color = "white";
+
+  const reverseColor = () => {
+    bodyStyle.backgroundColor =
+      bodyStyle.backgroundColor === black ? "white" : black;
+    bodyStyle.color = bodyStyle.color === "white" ? black : "white";
+  };
+
+  setInterval(reverseColor, 500);
+}
+function setAlarm() {
+  const countDownOnPage = () => {
+    document.querySelector(
+      "#timeRemaining"
+    ).textContent = `Time Remaining: ${min.toString().padStart(2, "0")}:${sec
+      .toString()
+      .padStart(2, "0")}`;
+
+    sec--;
+    if (min === 0 && sec <= 0) {
```

Comment:
It is expected that the alarm will ring when the remaining time is 0, but instead the alarm rings when remaining time is 1. So, the condition in here should be `if (min === 0 && sec < 0)`